### PR TITLE
feat: Speed up automated tests

### PR DESCRIPTION
**Pull request by kgpayne**

### CODE REVIEW

**Comment by kgpayne** [comment]
@pnadolny13 as part of speeding up tests, this also implements the `tables` config option as an array of strings 🙂

**Comment by kgpayne** [comment]
This is probably as fast as we'll get. Takes tests from ~20mins down to ~2m 🚀

**Comment by kgpayne** [comment]
Depends on [sdk#1399](https://github.com/meltano/sdk/pull/1399)

File: pyproject.toml [hunk 1]
```diff
@@ -16,9 +16,9 @@ packages = [
 [tool.poetry.dependencies]
 python = ">=3.7.1,<3.12"
 requests = "^2.25.1"
-singer-sdk = {extras = ["testing"], version = ">=0.22.1"}
-# use main until SDK 0.20.0
-# singer-sdk = {git = "https://github.com/meltano/sdk.git", rev = "main", extras=["testing"]}
+# singer-sdk = {extras = ["testing"], version = ">=0.22.1"}
+# TODO: remove before merge
+singer-sdk = {git = "https://github.com/meltano/sdk.git", rev = "feat-handle-sync-abort", extras=["testing"]}
```
Comment (edgarrmondragon) [inline pyproject.toml:21]:
```suggestion
singer-sdk = {git = "https://github.com/meltano/sdk.git", rev = "main", extras=["testing"]}
```

File: tap_snowflake/tap.py [hunk 2]
```diff
@@ -52,6 +52,11 @@ class TapSnowflake(SQLTap):
             th.StringType,
             description="The initial role for the session.",
         ),
+        th.Property(
+            "tables",
+            th.ArrayType(th.StringType),
+            description="An array of the table names that you want to sync. The table names should be fully qualified, including schema and table name. NOTE: this limits discovery to the tables specified, for performance reasons. Do not specify `tables` to discover entire available catalog.",  # noqa: E501
```
Comment (edgarrmondragon) [inline tap_snowflake/tap.py:58]:
```suggestion
            description=(
                "An array of the table names that you want to sync. The table names "
                "should be fully qualified, including schema and table name. "
                "NOTE: this limits discovery to the tables specified, for performance "
                "reasons. Do not specify `tables` if you intend to discover the entire "
                "available catalog."
            ), 
```

**Review by edgarrmondragon — APPROVED** [review]

**Comment by edgarrmondragon** [comment]
@kgpayne I've pushed changes to use the SDK `0.23.0`. Feel free to merge 😁

**Pull request merged**